### ISSUE TITLE
Building my first extension missing test result in failure

### ISSUE THREAD

**Issue by maxandersen** [quarkusio/quarkus#37216]
### Describe the bug

running:

`mvn io.quarkus.platform:quarkus-maven-plugin:3.5.2:create-extension -N -DgroupId=io.quarkiverse.code-server -DextensionId=code-server -DwithoutTests`

as documented in https://quarkus.io/guides/building-my-first-extension results in the project build failing:

```
mvn install
Using maven at '/Users/manderse/.sdkman/candidates/mvnd/current/bin/mvnd' to run buildFile '/Users/manderse/code/quarkus-code-server/pom.xml':
Nov 20, 2023 1:09:00 PM org.jline.utils.Log logr
WARNING: Unable to create a system terminal, creating a dumb terminal (enable debug logging for more information)
[main] WARN org.jline - Unable to create a system terminal, creating a dumb terminal (enable debug logging for more information)
[INFO] Processing build on daemon 6aea3359
[INFO] Scanning for projects...
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[ERROR] Child module /Users/manderse/code/quarkus-code-server/integration-tests of /Users/manderse/code/quarkus-code-server/pom.xml does not exist @
 @
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]
[ERROR]   The project io.quarkiverse.code-server:quarkus-code-server-parent:999-SNAPSHOT (/Users/manderse/code/quarkus-code-server/pom.xml) has 1 error
[ERROR]     Child module /Users/manderse/code/quarkus-code-server/integration-tests of /Users/manderse/code/quarkus-code-server/pom.xml does not exist
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
```

### Expected behavior

that getting started docs does not suggest create project that does not build :)

**Comment by maxandersen** [comment]
also the generated app is using 999-SNAPSHOT but docs says 1.0.0-SNAPSHOT

**Comment by gsmet** [comment]
I fixed a few broken things there already lately. I will have a look.

**Comment by gsmet** [comment]
I would argue that you shouldn't be surprised that your extension is broken if you don't have tests in it :]

**Comment by maxandersen** [comment]
@gsmet totally - hence I was surprised we didn't generated test by default.


Other thing that we should have is `-DextensionDescription="Do something useful."` as otherwise you create extensions that has quarkiverse description :)

**Comment by gsmet** [comment]
Yeah so FYI, I have on my TODO to improve this guide with some Quarkiverse stuff (or to write a specific guide targeting Quarkiverse).

We need the Quarkiverse way to be the easiest one.

**Comment by holly-cummins** [comment]
> Yeah so FYI, I have on my TODO to improve this guide with some Quarkiverse stuff (or to write a specific guide targeting Quarkiverse).
> 
> We need the Quarkiverse way to be the easiest one.

There's also https://hub.quarkiverse.io, which is I think a natural place for some? most? of the quarkiverse content, but we need to improve the cross-linking back and forth.

**Comment by gsmet** [comment]
@holly-cummins yeah, makes sense. I'll take your comment into account.

**Comment by holly-cummins** [comment]
Aha! I *knew* I'd tripped across the opposite problem, but I couldn't find any trace of it. It's this issue: https://github.com/quarkusio/quarkus/issues/29158. So I still can't find the change that did it, but it looks like #29158 was fixed (but not closed), and that created #37216?

**Comment by gsmet** [comment]
I created https://github.com/quarkusio/quarkus/pull/44141 to fix it.
